diff --git a/conda/activate.py b/conda/activate.py
index 0d28ecfd290..0fcb5c59e25 100644
--- a/conda/activate.py
+++ b/conda/activate.py
@@ -37,7 +37,6 @@
     CONDA_ENV_VARS_UNSET_VAR,
     PACKAGE_ENV_VARS_DIR,
     PREFIX_STATE_FILE,
-    RESERVED_ENV_NAMES,
     RESERVED_ENV_VARS,
 )
 from .base.context import context, locate_prefix_by_name
@@ -335,8 +334,6 @@ def _build_activate_stack(self, env_name_or_prefix, stack):
                 from .exceptions import EnvironmentLocationNotFound
 
                 raise EnvironmentLocationNotFound(prefix)
-        elif env_name_or_prefix in (RESERVED_ENV_NAMES):
-            prefix = context.root_prefix
         else:
             prefix = locate_prefix_by_name(env_name_or_prefix)
 
@@ -1000,6 +997,20 @@ def _hook_preamble(self) -> None:
         pass
 
 
+class CmdExeRunActivator(CmdExeActivator):
+    # CmdExeActivator writes an .env file by default; let's force in-memory output here
+    # so that we can embed the activation output directly into our wrapper script.
+    tempfile_extension = None
+
+    unset_var_tmpl = 'SET "%s="'
+    export_var_tmpl = 'SET "%s=%s"'
+    path_var_tmpl = export_var_tmpl
+    set_var_tmpl = export_var_tmpl
+    # If any of these calls to the activation hook scripts fail, we want
+    # to exit the wrapper immediately and abort `conda run` right away.
+    run_script_tmpl = 'CALL "%s"\nIF %%ERRORLEVEL%% NEQ 0 EXIT /b %%ERRORLEVEL%%'
+
+
 class FishActivator(_Activator):
     pathsep_join = '" "'.join
     sep = "/"
@@ -1119,6 +1130,7 @@ def _yield_commands(self, cmds_dict):
     "tcsh": CshActivator,
     "xonsh": XonshActivator,
     "cmd.exe": CmdExeActivator,
+    "cmd.exe.run": CmdExeRunActivator,
     "fish": FishActivator,
     "powershell": PowerShellActivator,
 }
diff --git a/conda/base/context.py b/conda/base/context.py
index c9c32769380..0b9bef0228c 100644
--- a/conda/base/context.py
+++ b/conda/base/context.py
@@ -2221,8 +2221,8 @@ def env_name(prefix: PathType) -> PathType | str | None:
 
 
 def locate_prefix_by_name(name: str, envs_dirs: PathsType | None = None) -> PathType:
-    """Find the location of a prefix given a conda env name.  If the location does not exist, an
-    error is raised.
+    """Find the location of a prefix given a conda env name. If the location does not exist, an
+    error is raised. If the prefix is the base or root env, the root prefix is returned.
     """
     if not name:
         raise ValueError("'name' cannot be empty.")
diff --git a/conda/utils.py b/conda/utils.py
index 7ca8738fc7d..8667a07aa31 100644
--- a/conda/utils.py
+++ b/conda/utils.py
@@ -15,6 +15,7 @@
 from typing import TYPE_CHECKING
 
 from . import CondaError
+from .activate import _build_activator_cls
 from .auxlib.compat import Utf8NamedTemporaryFile, shlex_split_unicode
 from .common.compat import isiterable, on_win
 from .common.url import path_to_url
@@ -221,14 +222,7 @@ def wrap_subprocess_call(
         multiline = True
     if on_win:
         comspec = get_comspec()  # fail early with KeyError if undefined
-        if dev_mode:
-            from . import CONDA_PACKAGE_ROOT
 
-            conda_bat = join(CONDA_PACKAGE_ROOT, "shell", "condabin", "conda.bat")
-        else:
-            conda_bat = environ.get(
-                "CONDA_BAT", abspath(join(root_prefix, "condabin", "conda.bat"))
-            )
         with Utf8NamedTemporaryFile(mode="w", suffix=".bat", delete=False) as fh:
             silencer = "" if debug_wrapper_scripts else "@"
             fh.write(f"{silencer}ECHO OFF\n")
@@ -256,7 +250,22 @@ def wrap_subprocess_call(
             # after all!
             # fh.write("@FOR /F \"tokens=100\" %%F IN ('chcp') DO @SET CONDA_OLD_CHCP=%%F\n")
             # fh.write('@chcp 65001>NUL\n')
-            fh.write(f'{silencer}CALL "{conda_bat}" activate "{prefix}"\n')
+
+            # We pursue activation inline here, which allows us to avoid
+            # spawning a `conda activate` process at wrapper runtime.
+            activator_cls = _build_activator_cls("cmd.exe.run")
+            activator_args = ["activate"]
+            if dev_mode:
+                activator_args.append("--dev")
+            activator_args.append(prefix)
+
+            activator = activator_cls(activator_args)
+            activator._parse_and_set_args()
+            activate_script = activator.activate()
+
+            for line in activate_script.splitlines():
+                fh.write(f"{silencer}{line}\n")
+
             fh.write(f"{silencer}IF %ERRORLEVEL% NEQ 0 EXIT /b %ERRORLEVEL%\n")
             if debug_wrapper_scripts:
                 fh.write("echo *** environment after *** 1>&2\n")
@@ -316,6 +325,9 @@ def wrap_subprocess_call(
             dev_arg = ""
             dev_args = []
         with Utf8NamedTemporaryFile(mode="w", delete=False) as fh:
+            # If any of these calls to the activation hook scripts fail, we want
+            # to exit the wrapper immediately and abort `conda run` right away.
+            fh.write("set -e\n")
             if dev_mode:
                 from . import CONDA_SOURCE_ROOT
 
@@ -325,9 +337,25 @@ def wrap_subprocess_call(
                 fh.write(">&2 echo '*** environment before ***'\n>&2 env\n")
                 fh.write(f'>&2 echo "$({hook_quoted})"\n')
             fh.write(f'eval "$({hook_quoted})"\n')
-            fh.write(f"conda activate {dev_arg} {quote_for_shell(prefix)}\n")
+
+            # We pursue activation inline here, which allows us to avoid
+            # spawning a `conda activate` process at wrapper runtime.
+            activator_cls = _build_activator_cls("posix")
+            activator_args = ["activate"]
+            if dev_mode:
+                activator_args.append("--dev")
+            activator_args.append(prefix)
+
+            activator = activator_cls(activator_args)
+            activator._parse_and_set_args()
+            activate_code = activator.activate()
+
+            fh.write(activate_code)
+
             if debug_wrapper_scripts:
                 fh.write(">&2 echo '*** environment after ***'\n>&2 env\n")
+            # Disable exit-on-error for the user's command so we can capture its exit code.
+            fh.write("set +e\n")
             if multiline:
                 # The ' '.join() is pointless since mutliline is only True when there's 1 arg
                 # still, if that were to change this would prevent breakage.
